compute_sdk/tests/unit/test_client.py

```diff
@@ -419,3 +429,35 @@ def test_version_mismatch_from_details(mocker, gcc, mock_response, dill_python):
             assert f"but worker used {worker_python}" in a[0]
         if worker_dill:
             assert f"/{worker_dill} (OS" in a[0]
+
+
+@pytest.mark.parametrize("num_runs", range(2, 10))
+def test_client_only_warns_version_mismatch_once(mocker, gcc, mock_response, num_runs):
```

Comment:
This test only checks that the version mismatch logging is done if there's an exception raised ("exception" in returned_task).

I think we should add another test (or modify this one with added returned_task mock without "exception") to ensure that the warning is done as per this new story?

Reply:
I changed the structure of the tests to handle this - most of the tests now test for success by default, with one test specifically to test that failure works too. Tests in general for this behavior should only have single reasons to fail now.